src/models/User.ts

```diff
@@ -6,14 +6,20 @@ interface User {
     password: string;
     registeredAt: Date;
     verified: boolean;
+    emailCode: string;
+    resetCode: string;
+    resetAt: number;
 }
 
 const userSchema = new Schema<User>({
-    name: String,
+    name: { type: String, required: true },
     email: { type: String, required: true },
-    password: String,
-    registeredAt: Date,
-    verified: Boolean,
+    password: { type: String, required: true },
+    registeredAt: { type: Date, required: true },
+    verified: { type: Boolean, required: true },
```

Comment:
Initially we made these not-required since users should be able to book via Discord without making a separate account due to existing verification setups, and it might be beneficial to create a user model for them even if the other fields aren't filled in.

Any thoughts on how we should handle the Discord use case? An example of where a user model without registration might be useful is if we wanted to ban a non-registered user from booking, and we could use a `bannedUntil` key on the user model to do it.